### DISCUSSION TITLE
torch.jit issue during import

### DISCUSSION THREAD

**Discussion by clamence** [kornia/kornia#1572] [Potential Issue] [answered]
### Describe the bug

When I try to import get_perspective_transform, warp_affine, I have the following error (Ubuntu 18.04, kornia 0.6.3, python 3.7.11)

```
    from kornia.geometry.transform.imgwarp import get_perspective_transform, warp_affine
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/kornia/__init__.py", line 10, in <module>
    from . import (
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/kornia/augmentation/__init__.py", line 52, in <module>
    from kornia.augmentation.container import AugmentationSequential, ImageSequential, PatchSequential, VideoSequential
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/kornia/augmentation/container/__init__.py", line 1, in <module>
    from kornia.augmentation.container.augment import AugmentationSequential
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/kornia/augmentation/container/augment.py", line 15, in <module>
    from kornia.geometry.boxes import Boxes
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/kornia/geometry/boxes.py", line 464, in <module>
    @torch.jit.script
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/torch/jit/_script.py", line 924, in script
    _compile_and_register_class(obj, _rcb, qualified_name)
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/torch/jit/_script.py", line 64, in _compile_and_register_class
    torch._C._jit_script_class_compile(qualified_name, ast, defaults, rcb)
RuntimeError: 
Tried to access nonexistent attribute or method '__call__' of type '__torch__.kornia.geometry.boxes.Boxes3D'. Did you forget to initialize an attribute in __init__()?:
  File "/home/ubuntu/anaconda3/envs/base/lib/python3.7/site-packages/kornia/geometry/boxes.py", line 601
        # Due to some torch.jit.script bug (at least <= 1.9), you need to pass all arguments to __init__ when
        # constructing the class from inside of a method.
        return cls(hexahedrons, raise_if_not_floating_point=False, mode=mode)
               ~~~ <--- HERE
```

### Reproduction steps

```bash
1.from kornia.geometry.transform.imgwarp import get_perspective_transform, warp_affine
```

### Expected behavior

No error during import

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.17

Python version: 3.7.11 (default, Jul 27 2021, 14:32:16)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-1065-azure-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 460.73.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.5
[pip3] pytorch3d==0.3.0
[pip3] torch==1.7.1
[pip3] torchvision==0.8.2
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.0.221             h6bb024c_0  
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py37h7f8727e_0  
[conda] mkl_fft                   1.3.1            py37hd3c417c_0  
[conda] mkl_random                1.2.2            py37h51133e4_0  
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.21.5                   pypi_0    pypi
[conda] numpy-base                1.20.3           py37h74d4b33_0  
[conda] pytorch                   1.7.1           py3.7_cuda11.0.221_cudnn8.0.5_0    pytorch
[conda] pytorch3d                 0.3.0                    pypi_0    pypi
[conda] torchvision               0.8.2                py37_cu110    pytorch
```


### Additional context

I installed kornia with no dependencies. Is it because I still have torch 1.7.1 ? Is it still supported ?

**Comment by edgarriba** [comment, marked as answer]
One way to know is you updating at least to 1.8.1. We made last mínimum version few releases ago.
Besides, you can just try
 `from kornia.geometry.transform import get_perspective_transform, warp_affine`
 or
`from kornia.geometry import get_perspective_transform, warp_affine`

**Comment by clamence** [comment]
Thank you for your answer, updating to 1.10 definitely worked !

**Comment by hengfei-wang** [comment]
Hi,

I have the same problem. I have updated my torch version to 1.12.1 (also tried 1.10.0). But the problem still exists. Help wanted.

Thanks

**Reply by edgarriba** [reply]
what's the kornia version ? ideally provide a way to reproduce

**Comment by hengfei-wang** [comment]
Hi, I have solved my problem. Even though it is strange, I did it in the following steps.

1. pip install pytorch3d (get related modules for pytorch3d, this version of pytorch3d is too old, will update in last step)
2. pip install -r requirements.txt (include kornia)
3. pip install 'git+https://github.com/facebookresearch/pytorch3d.git@stable' (update pytorch3d version to the lastest)
4. run demo successfully
